### PR TITLE
fix ignored annotations with namespace alias

### PR DESCRIPTION
**Pull request by dbu**
when i set a namespace to ignored but use a namespace alias, there was no check if the class actually exists. instead of ignoring, we ended up with an exception that the annotation could not be autoloaded.

### CODE REVIEW

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 1]
```diff
@@ -705,10 +705,11 @@ private function Annotation()
                     }
                 }
             } elseif (isset($this->imports[$loweredAlias])) {
-                $found = true;
                 $name  = (false !== $pos)
                     ? $this->imports[$loweredAlias] . substr($name, $pos)
                     : $this->imports[$loweredAlias];
+                $name = ltrim($name,'\\');
```
Comment (dbu) [inline lib/Doctrine/Common/Annotations/DocParser.php]: this is duplicated below, but we need it here to make $this->classExists work

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 2]
```diff
@@ -940,10 +941,10 @@ private function Constant()
                     break;
 
                 case isset($this->imports[$loweredAlias]):
-                    $found     = true;
                     $className = (false !== $pos)
                         ? $this->imports[$loweredAlias] . substr($className, $pos)
                         : $this->imports[$loweredAlias];
+                    $found     = class_exists($className) || interface_exists($className);
```
Comment (dbu) [inline lib/Doctrine/Common/Annotations/DocParser.php]: similar issue with constants. i found no tests for this but think we should handle it the same way? i can also remove this line again if you prefer.

File: tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php [hunk 3]
```diff
@@ -112,6 +113,21 @@ public function testPropertyAnnotationIsIgnored()
         self::assertEmpty($reader->getPropertyAnnotations($ref->getProperty('property')));
     }
 
+    /**
+     * @group 45
```
Comment (dbu) [inline tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: not sure what this is for, but copied from the other ignored annotations tests.

**Comment by dbu** [comment]
travis error is unrelated, see #245

File: tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php (hunk 3 above)
Comment (Ocramius) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: This is for the issue ID: this would be 244 in this case

File: lib/Doctrine/Common/Annotations/DocParser.php (hunk 1 above)
Comment (Ocramius) [reply lib/Doctrine/Common/Annotations/DocParser.php]: Please avoid overwriting the variable: wrap the previous expression, if possible

File: lib/Doctrine/Common/Annotations/DocParser.php (hunk 2 above)
Comment (Ocramius) [reply lib/Doctrine/Common/Annotations/DocParser.php]: Could you rephrase this point? I didn't understand the comment.

Comment (Ocramius) [reply lib/Doctrine/Common/Annotations/DocParser.php]: This addition requires additional tests for:

 * class not found
 * interface not found
 * class found
 * interface found

Comment (dbu) [reply lib/Doctrine/Common/Annotations/DocParser.php]: i will revert this change as i am not sure about it. i suspect the same issue to be here for the constants handling as we have for annotations handling, because it also handles namespace alias separately.

File: tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php (hunk 3 above)
Comment (dbu) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: ah, that makes sense :-D 
fixing that number

File: lib/Doctrine/Common/Annotations/DocParser.php (hunk 1 above)
Comment (dbu) [reply lib/Doctrine/Common/Annotations/DocParser.php]: did it differently, okay like this?

**Comment by dbu** [comment]
okay, updated to the feedback.

**Comment by Majkl578** [comment]
As per existing branching rules (which are still not formalized), this should not target 1.6. :/

**Comment by alcaeus** [comment]
@Majkl578 we haven't discussed this, and in an effort to do less cherry-picking and more merging I suggested that the PR be opened against 1.6, as #243 was previously targeting master. With that in mind, I'd no longer ask @dbu to retarget the PR but rather take care of this ourselves when merging the PR and preparing for a new release.

**Comment by dbu** [comment]
hey, i don't mind doing the PR against the right branch again, so long as you can agree on which branch that should be ;-)

**Comment by dbu** [comment]
can i do something to get this merged into the appropriate branch?

**Comment by alcaeus** [comment]
@Majkl578 go ahead and merge this to whatever branch you deem appropriate - I'll let you take care of ensuring this ends up in a 1.6 patch release.

**Comment by greg0ire** [comment]
@dbu the tests no longer pass, please have a look.

**Comment by dbu** [comment]
thanks for wrapping this up @alcaeus ! i am currently a bit busy and did not get around to do it.
